index.html

tongrhj opened the pull request.

tongrhj committed:
Merge pull request #1 from caalberts/master

Merge

tongrhj committed:
Merge remote-tracking branch 'caalberts/master'

tongrhj committed:
Implemented Index.html

tongrhj committed:
Merge remote-tracking branch 'origin/master'

tongrhj committed:
Merge remote-tracking branch 'caalberts/master'

tongrhj committed:
Renamed flat to type and cleared up fetch code

caalberts commented on src/app.js:
```diff
@@ -5,69 +5,74 @@ import bodyParser from 'body-parser'
+app.use(express.static('public'))
+
+
+// const dbUri = 'mongodb://' +
```
Please uncomment, it's causing git conflict

caalberts commented on src/app.js:
```diff
@@ -5,69 +5,74 @@ import bodyParser from 'body-parser'
-    }
-  })
-})
+// app.get('/flats', function (req, res) {
```
Please uncomment, it's causing git conflict

caalberts commented on src/app.js:
```diff
@@ -5,69 +5,74 @@ import bodyParser from 'body-parser'
-    res.status(201).json(newTownFlat)
-  })
-})
+// app.post('/flats', function (req, res) {
```
Please uncomment, it's causing git conflict

caalberts commented on views/index.html:
```diff
@@ -0,0 +1,34 @@
+
+    </script>
+  </body>
+</html>
```
index.html looks good :+1: :smile: 

should the directory be `public` instead of `views`, because in app.js we are serving `app.use(express.static('public'))`?

caalberts commented on src/index.js:
```diff
@@ -0,0 +1,24 @@
+function fetchFromURL (town, type) {
+  const url = '/flats?town=' + town + '&type=' + type
+  console.log('Fetching: ' + url)
+  window.fetch(url)
```
just `fetch()` instead of `window.fetch()`

caalberts commented on src/app.js:
```diff
@@ -5,69 +5,74 @@ import bodyParser from 'body-parser'
-  type: String,
-  data: Array
-})
+app.use(express.static('dist'))
```
`dist` folder is not static, it contains the server codes

tongrhj committed:
Moved index.js and html to public folder, uncommented app.js to avoid merge conflicts

tongrhj committed:
Merge remote-tracking branch 'caalberts/master'

tongrhj committed:
Cache node modules for faster travis-ci checks

tongrhj committed:
Changed fetch to window.fetch for maximum Standard compatibility

tongrhj commented:
@caalberts: It has to be window.fetch rather than fetch for Standard compatibility! Is there another way?

I've moved the index files to their own public folder, which gets copied to the dist.

_To do: style the information received and display it in a visually engaging way_

caalberts commented on .travis.yml:
```diff
@@ -3,3 +3,6 @@ node_js:
   - 5
 after_script:
   # - npm run codecov
+cache:
```
cool! i didn't know can do this

caalberts commented:
@tongrhj sure if window.fetch() works too then i think it's fine. :smile: 

review app is on https://hdb-resale-data-pr-2.herokuapp.com/

caalberts commented on package.json:
```diff
@@ -4,8 +4,9 @@
-    "start": "npm run build && npm run start-server",
+    "start": "npm run build && npm run build-index && npm run start-server",
     "start-server": "node dist/server.js",
+    "build-index": "cp public/index.html dist/public/index.html && babel public --out-dir dist/public --presets es2015",
```
Heroku build is failing. Logs below. I think need to `mkdir dist/public` before `cp ... dist/public/index.html`

```
2015-12-03T14:51:11.644900+00:00 app[web.1]: src/server.js -> dist/server.js
2015-12-03T14:51:11.691153+00:00 app[web.1]: src/util/hdb.js -> dist/util/hdb.js
2015-12-03T14:51:11.631426+00:00 app[web.1]: src/app.js -> dist/app.js
2015-12-03T14:51:12.235359+00:00 app[web.1]: npm ERR! argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "run" "build-index"
2015-12-03T14:51:12.258556+00:00 app[web.1]: npm ERR! Linux 3.13.0-66-generic
2015-12-03T14:51:12.218076+00:00 app[web.1]: 
2015-12-03T14:51:12.218081+00:00 app[web.1]: > hdb-resale-data@1.0.0 build-index /app
2015-12-03T14:51:12.226270+00:00 app[web.1]: cp: cannot create regular file 'dist/public/index.html': No such file or directory
2015-12-03T14:51:12.235771+00:00 app[web.1]: npm ERR! node v5.0.0
2015-12-03T14:51:12.218083+00:00 app[web.1]: 
2015-12-03T14:51:12.235031+00:00 app[web.1]: npm ERR! Linux 3.13.0-66-generic
2015-12-03T14:51:12.218082+00:00 app[web.1]: > cp public/index.html dist/public/index.html && babel public --out-dir dist/public --presets es2015
2015-12-03T14:51:12.236287+00:00 app[web.1]: npm ERR! npm  v3.3.6
2015-12-03T14:51:12.230842+00:00 app[web.1]: 
2015-12-03T14:51:12.236664+00:00 app[web.1]: npm ERR! Exit status 1
```

tongrhj committed:
Attempt to fix the fetch URL

Problem: Was sending a POST request

Attempted Solutions:
1. Change Prophesize type from submit to generic button
2. Use /* global fetch */ to avoid window.fetch
3. Reorganised public folder out of dist

tongrhj committed:
Full URL fix

tongrhj committed:
Fixed Base URL function

tongrhj committed:
Reverted to relative url string
